aesara/tensor/random/basic.py

```diff
@@ -643,20 +643,17 @@ class CategoricalRV(RandomVariable):
     @classmethod
     def rng_fn(cls, rng, p, size):
         if size is None:
-            size = ()
-
-        size = tuple(np.atleast_1d(size))
-        ind_shape = p.shape[:-1]
-
-        if len(ind_shape) > 0:
-            if len(size) > 0 and size[-len(ind_shape) :] != ind_shape:
-                raise ValueError("Parameters shape and size do not match.")
-
-            samples_shape = size[: -len(ind_shape)] + ind_shape
+            size = p.shape[:-1]
         else:
-            samples_shape = size
-
-        unif_samples = rng.uniform(size=samples_shape)
+            # Check that `size` does not define a shape that would be broadcasted
```

Comment:
In a previous iteration, I simply set `p=np.broadcast_to(p, size + p.shape[-1:])`, and let that capture broadcast issues, but calling `cumsum` on the broadcasted array seems to be considerably slower, so I opted for the explicit check here.

Reply:
FYI: I used that `np.searchsorted` approach only for the convenience, so we should definitely consider alternative approaches when/if they address other issues (e.g. code complexity, performance, etc.) as well.

Reply:
I don't see anything obvious with that approach. This PR already simplifies logic a bit so I am happy with the state of affairs.